pelicun/base.py

```diff
@@ -328,6 +330,10 @@ def __init__(
         self.reset_log_strings()
         control_warnings()
 
+        # Set sys.excepthook to handle uncaught exceptions
+        # https://docs.python.org/3/library/sys.html#sys.excepthook
+        sys.excepthook = self.log_exception
```

Comment:
What happens if we create multiple Logger objects? Would the last one's log_exception overwrite the previous ones in sys.excepthook? That's how it looks now. If that is indeed the case, I wonder if we could do something about it.
I don't think we will be able to pick the right logger when an exception occurs, so I suggest printing the exception in every active log file instead. One way to do that would be to create the hook outside of the logger class and link a general log_exception function there. Then, whenever a new Logger object is created, we just need to make sure to "inform" the log_exception function that it needs to send a message to this new logger when an exception occurs. This could be achieved with something as simple as a global variable that collects the active logger objects, but perhaps you have a better idea.
In any case, it doesn't seem too hard to generalize the current implementation so that we can keep supporting multiple, assessment-specific loggers in Pelicun.
Let me know your thoughts.

Reply:
Thank you Adam,
That is a good point. Indeed, the prior implementation would only log the exception via the last instantiated logger object. In the case of multiple loggers, it wouldn't be immediately clear where the exception would be documented. I updated the implementation to log the exception in all instantiated loggers, based on your description. b51d5329

Reply:
Excellent!